spec/requests/api/v1/error_handling_spec.rb

```diff
@@ -0,0 +1,88 @@
+describe 'Error handling', type: :request do
+  context "when request doesn't contain required parameters" do
+    it 'responds with 400 BadRequest' do
+      post '/api/v1/locations', params: {}, headers: default_headers
+
+      expect(response).to have_http_status(:bad_request)
+      error = json_response['errors'].first
+      expect(error['title']).to eq('Bad Request')
+      expect(error['detail']).to match('param is missing or the value is empty: location')
+    end
+  end
+
+  context "when ActiveRecord can't find requested record" do
+    it 'responds with 404 NotFound' do
+      get '/api/v1/locations/0', headers: default_headers
+
+      expect(response).to have_http_status(:not_found)
+      expect(json_response['errors'].first['title']).to eq('Not found')
+      expect(json_response['errors'].first['detail']).to eq('Resource not found')
+    end
+  end
+
+  context 'when client is not authorized to perform requested action' do
+    it 'responds with 403 Forbidden' do
+      loc = Location.create(latitude: -1, longitude: -1)
+      get "/api/v1/locations/#{loc.id}", headers: default_headers
+
+      expect(response).to have_http_status(:forbidden)
+      expect(json_response['errors'].first['title']).to eq('Forbidden')
+      expect(json_response['errors'].first['detail'])
+        .to eq('You are not allowed to perform this action')
+    end
+  end
+
+  context 'when request contains unpermitted sort params' do
```

Comment:
This seems like query builder specific logic, do we really want to test this in this gem?

As far as the behavior is concerned I would rather not silently swallow requests like this since they mask integration errors from the backend. Maybe a  400 response code + bugsnag notify could be a middle ground?

Reply:
> This seems like query builder specific logic, do we really want to test this in this gem?

What do you mean?
This spec checks if [query builder error is handled properly](https://github.com/infinum/infinum-json-api-setup/pull/24/files#diff-3517d53722e46b93c1323017cb8c954ab6a89330a88779f006bab4ec40554e3eR24) ?

> As far as the behavior is concerned I would rather not silently swallow requests like this since they mask integration errors from the backend. Maybe a 400 response code + bugsnag notify could be a middle ground?

Sounds good to me 👍

Reply:
If we follow this logic, aren't we by extension also 'testing' the `Pundit` gem?

Reply:
The same answer, it check is [Pundit error is handled properly](https://github.com/infinum/infinum-json-api-setup/pull/24/files#diff-3517d53722e46b93c1323017cb8c954ab6a89330a88779f006bab4ec40554e3eR17)

Reply:
athough, [this message](https://github.com/infinum/infinum-json-api-setup/pull/24/files#diff-3517d53722e46b93c1323017cb8c954ab6a89330a88779f006bab4ec40554e3eR18) looks quite dubious, I missed it in original implementation and would like to move it into locale file

Reply:
My question was directed more to @uncoverd, as I understood his point as whether we should rely on query builder to raise an error. I think it's ok to do that.

Reply:
So we're prepping a template that will be used with our preferred stack as-is and we're okay with some coupling in the integration spec, understood 👍

Reply:
@djordjelacmanovic I think we are, I just wasn't confident enough in the reasoning behind the approach to voice it 😄 I checked out the repo locally now and it makes much more sense than plain diffs did before 🙈 If we define the behaviour we should test it and this is an integration spec so it should be as real as we can make it 👍